crates/provider-sdk/src/error.rs

```diff
@@ -3,9 +3,6 @@
 pub type InvocationResult<T> = Result<T, InvocationError>;
 pub type ProviderInitResult<T> = Result<T, ProviderInitError>;
 
-/// Result form for [`ProviderHandler`] trait methods
-pub type ProviderOperationResult<T> = Result<T, Box<dyn std::error::Error + Send + Sync>>;
-
```

Comment:
For this change from basically the standard Result -> an anyhow::Result, what's the motivation? I don't actually feel dedicated to either route, I originally remember talking with Victor about using this form to be a little more flexible

Reply:
I actually prefer anyhow to boxed dynner errors. If anything, I'd love for us to have a generic `E` if we can, but I can live without it

Reply:
Agree

Reply:
@brooksmtownsend the effect here is actually the opposite, the error is generic `E`, just like @thomastaylor312 described in https://github.com/wasmCloud/wasmCloud/pull/1817#discussion_r1554275395
`anyhow::Error` is merely the default - developers are free to choose any error type they wish